### PR TITLE
chore(Button): updating IconButton's variants' icon sizes

### PR DESCRIPTION
**Pull request by LinKCoding**
### Overview
IconButtons use icons that currently size depending on the parent container using `"calc(100% - 14)"`
But for some reason this calculation seems to be done using the width/height of `::after` which can be marginally larger (e.g. 40.8 when it should be exactly 40)
image
^ on the prod version of storybook, it comes out to 40 🤷 

Regardless, the calculation of `-14px` should actually be `-18px` to enable a padding of 8px all around.

To make the sizing more explicit, concrete values were used:
normal: 24, <- from figma
small: 16, <- from figma
large: 40, <- assuming since currently a large IconButton is 56 and 56 - 16 = 40, the figma will need to be updated to reflect this size. Confirmed with Stacey

### PR Checklist

- [ ] Related to designs: https://www.figma.com/design/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=1106-90&m=dev
- [ ] Related to JIRA ticket: https://codecademy.atlassian.net/browse/GM-702 
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change
- [ ] This PR includes testing instructions tests for the code change
- [ ] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

1. Go to https://666b16bc273aa9527803b3d5--gamut-preview.netlify.app/?path=/docs/atoms-button--icon-button 
2. Inspect the IconButton element's variants, check that:
  a. normal: parent container is 40x40 and the icon is 24x24
  b. small: parent container is 32x32 and the icon is 16x16
  c. large: parent container is 56x56 and the icon is 40x40
---
3. Go to the Monolith Preview and `/admin/concessions`
4. Check that edit icon buttons (they're small) and should render correctly (32x32 for parent, 16x16 for the icon)
--- 
5. Go to Mono Preview (Do NOT login)
6. This main page should be the signup page with a carousel and a pause button, inspect the pause button
image
7. This is also a small variant and should render correctly (32x32 for parent, 16x16 for the icon)
8. ...
9. Profit!

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/37875)  | [Monolith Env](https://pr-37875-monolith.dev-eks.codecademy.com/) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/6644)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-6644)   |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.22.4-alpha.163f33.0<br/>@codecademy/gamut-kit@0.6.416-alpha.163f33.0<br/>@codecademy/styleguide@66.25.1-alpha.163f33.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://666b16bc273aa9527803b3d5--gamut-preview.netlify.app>https://666b16bc273aa9527803b3d5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/666b16bc273aa9527803b3d5>Deploy Logs</a></p>

File: packages/gamut/src/Button/IconButton.tsx [hunk 1]
```diff
@@ -43,13 +43,20 @@ export const IconButton = forwardRef<ButtonBaseElements, IconButtonProps>(
 
     const trueAriaLabel = ariaLabel ?? firstWord;
 
+    const size = props.size || 'normal';
```
Comment (dreamwasp) [inline packages/gamut/src/Button/IconButton.tsx:46]: nit: can you add this to the destructed props above?

File: packages/gamut/src/Button/IconButton.tsx [hunk 2]
```diff
@@ -43,13 +43,20 @@ export const IconButton = forwardRef<ButtonBaseElements, IconButtonProps>(
...
+    const iconSizeMapping = {
+      normal: 24,
+      small: 16,
+      large: 40,
```
Comment (dreamwasp) [inline packages/gamut/src/Button/IconButton.tsx:50]: nit: add to styles folder, i think it makes more sense to be there

**Review by dreamwasp — APPROVED** [review]
i have the tiniest of nits but this looks perfect! 🔥

File: packages/gamut/src/Button/IconButton.tsx [hunk 3]
```diff
@@ -58,11 +65,7 @@ export const IconButton = forwardRef<ButtonBaseElements, IconButtonProps>(
           aria-describedby={tipId}
           aria-label={trueAriaLabel}
         >
-          <Icon
-            width="calc(100% - 14px)"
-            height="calc(100% - 14px)"
-            aria-hidden
-          />
+          <Icon size={iconSize} />
```
Comment (dreamwasp) [inline packages/gamut/src/Button/IconButton.tsx:68]: can you re-add aria-hidden here?

**Review by dreamwasp — CHANGES_REQUESTED** [review]
just blocking till aria-hidden is added back, otherwise looks great!

File: packages/gamut/src/Button/IconButton.tsx (hunk 3 above)
Comment (LinKCoding) [reply packages/gamut/src/Button/IconButton.tsx:68]: woops, thanks!

File: packages/gamut/src/Button/IconButton.tsx (hunk 1 above)
Comment (LinKCoding) [reply packages/gamut/src/Button/IconButton.tsx]: can do!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.22.5-alpha.f134e0.0<br/>@codecademy/gamut-kit@0.6.417-alpha.f134e0.0<br/>@codecademy/styleguide@66.25.2-alpha.f134e0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6675a2a6803efaa9e7dbf1b0--gamut-preview.netlify.app>https://6675a2a6803efaa9e7dbf1b0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6675a2a6803efaa9e7dbf1b0>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.22.6-alpha.636d8b.0<br/>@codecademy/gamut-kit@0.6.418-alpha.636d8b.0<br/>@codecademy/styleguide@66.26.1-alpha.636d8b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6675dfd6af420307ed07c73c--gamut-preview.netlify.app>https://6675dfd6af420307ed07c73c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6675dfd6af420307ed07c73c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.22.6-alpha.636d8b.0<br/>@codecademy/gamut-kit@0.6.418-alpha.636d8b.0<br/>@codecademy/styleguide@66.26.1-alpha.636d8b.0<br/></pre>

**Comment by LinKCoding** [comment]
@dreamwasp ah wait -- I remember why I took out `aria-hidden`, Icon should come with aria-hidden true already, so I just removed it here because it wasn't needed. 

I added it to IconButtonBase weirdly 🤔 (and it's breaking some tests lol) 

gonna fix that now, but we can also disc on Monday. Thanks!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.22.6-alpha.7a40b3.0<br/>@codecademy/gamut-kit@0.6.418-alpha.7a40b3.0<br/>@codecademy/styleguide@66.26.1-alpha.7a40b3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6675ee49b8e00cf6bd9863e6--gamut-preview.netlify.app>https://6675ee49b8e00cf6bd9863e6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6675ee49b8e00cf6bd9863e6>Deploy Logs</a></p>

**Comment by dreamwasp** [comment]
> @dreamwasp ah wait -- I remember why I took out `aria-hidden`, Icon should come with aria-hidden true already, so I just removed it here because it wasn't needed.
> 
> I added it to IconButtonBase weirdly 🤔 (and it's breaking some tests lol)
> 
> gonna fix that now, but we can also disc on Monday. Thanks!

oh duh!!! you're totally right. this is perfect ✨

**Review by dreamwasp — APPROVED** [review]

**Pull request merged**